src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -301,6 +325,17 @@ Image& Frame::image() const
     return *impl->m_image.get();
 }
 
+Image& Frame::ref_image() const
+{
+    return *impl->m_ref_image.get();
```

Comment:
I would return an `Image*` given that the ref image might not be there (and then document that behavior in the header file).

Reply:
Right, and then we don't need to dereference what would potentially be a `nullptr` as you've mentioned above.

Thank you for this advice!